sotodlib/tod_ops/interp_drop.py

```diff
@@ -0,0 +1,100 @@
+import numpy as np
+from sotodlib import core
+from so3g.proj import Ranges
+
+def interp_drop_array(arr, drop_idxes):
+    """
+    Function to interpolate numpy.array.
+    Insert the mean of left and right samples at the dropped indexes
+    
+    Args:
+        arr: array contains data drops.
+        drop_idxes: Indexes of dropped samples.
+        
+    Return:
+        new_arr: array that drops are filled.
+    """
+    data_dim = len(np.shape(arr))
+    new_arr = np.copy(arr)
+    
+    for  i in range(len(drop_idxes)):
+        drop_idx = drop_idxes[i] + i
+        if data_dim == 1:
+            # Such as azimuth or elevation
+            inserted_data = (new_arr[drop_idx-1] + new_arr[drop_idx])/2.
+            new_arr = np.insert(new_arr, drop_idx, inserted_data)
+            
+        elif data_dim == 2:
+            inserted_data = (new_arr[:, drop_idx-1] + new_arr[:, drop_idx])/2.
+            new_arr = np.insert(new_arr, drop_idx, inserted_data, axis=1)
+        else:
+            raise ValueError(f'Unsupported data dimensions ({field})')
+ 
+    return new_arr
+
+def interp_drop_single_aman(aman, drop_idxes):
+    """
+    Function to interpolate axismanager.
+    Insert the mean of left and right samples at the dropped indexes.
+    
+    Args:
+        aman: axismanager object.
+        drop_idxes: Indexes of dropped samples.
+    Return:
+        new_aman: axismanager object that drops are filled.
+    """
+    new_samps = core.OffsetAxis(name='samps', 
+                                count=aman.samps.count + len(drop_idxes), 
+                                offset=aman.samps.offset)
+    
+    other_axes = []
+    for axis_name, axis_obj in aman._axes.items():
+        if axis_name != 'samps':
```

Comment:
Please make the name of the target axis a kwarg... or at least mention the interpolation is along the 'samps' axis.